// @masoni/components/dist/esm2020/lib/sub-elements/label.component.mjs
import { CommonModule } from '@angular/common';
import { Component, Input, NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class MasLabel {
    get identifier() {
        return this._identifier;
    }
    set identifier(value) {
        this._identifier = value;
    }
    constructor() {
        this.optional = false;
        this.infoIcon = false;
    }
    ngOnInit() { }
}
MasLabel.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: MasLabel, deps: [], target: i0.ɵɵFactoryTarget.Component });
MasLabel.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.2.9", type: MasLabel, selector: "mas-label", inputs: { identifier: "identifier", optional: "optional", infoIcon: "infoIcon" }, host: { classAttribute: "mas-label" }, ngImport: i0, template: `
    <label [for]="identifier" class="mas-label-wrapper">
      <ng-content></ng-content>
      <i *ngIf="infoIcon" class="mas-question-outlined mas-label-info-icon"></i>
      <span *ngIf="optional" class="mas-label-optional">(optional)</span>
    </label>
  `, isInline: true, dependencies: [{ kind: "directive", type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: MasLabel, decorators: [{
            type: Component,
            args: [{
                    selector: 'mas-label',
                    host: {
                        class: 'mas-label',
                    },
                    template: `
    <label [for]="identifier" class="mas-label-wrapper">
      <ng-content></ng-content>
      <i *ngIf="infoIcon" class="mas-question-outlined mas-label-info-icon"></i>
      <span *ngIf="optional" class="mas-label-optional">(optional)</span>
    </label>
  `,
                }]
        }], ctorParameters: function () { return []; }, propDecorators: { identifier: [{
                type: Input
            }], optional: [{
                type: Input
            }], infoIcon: [{
                type: Input
            }] } });
export class MasLabelModule {
}
MasLabelModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: MasLabelModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MasLabelModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.9", ngImport: i0, type: MasLabelModule, declarations: [MasLabel], imports: [CommonModule, FormsModule], exports: [MasLabel] });
MasLabelModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: MasLabelModule, imports: [CommonModule, FormsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: MasLabelModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, FormsModule],
                    exports: [MasLabel],
                    declarations: [MasLabel],
                }]
        }] });
