// @masoni/components/dist/esm2020/lib/select/select.component.mjs
import { Component, ContentChildren, EventEmitter, Input, Output, TemplateRef, HostListener, } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import '@masoni/css-select/dist/index.css';
import { MasOption } from './option.component';
import { Subject, defer, merge, startWith, switchMap, take, takeUntil } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class MasSelect {
    set value(value) {
        this._value = value;
    }
    get value() {
        if (this.empty) {
            return '';
        }
        return this.options.filter((option) => option.selected)[0].value;
    }
    get placeholder() {
        return this._placeholder;
    }
    set placeholder(value) {
        this._placeholder = value;
    }
    get panelOpen() {
        return this._panelOpen;
    }
    get triggerValue() {
        if (this.empty) {
            return '';
        }
        return this.selected.option?.text;
    }
    get empty() {
        return !this.options || this.selected.isSelected === false;
    }
    ngAfterContentInit() {
        this.options.changes.pipe(startWith(null), takeUntil(this._destroy)).subscribe(() => {
            this._resetOptions();
        });
    }
    constructor(_ngZone, eRef) {
        this._ngZone = _ngZone;
        this.eRef = eRef;
        this._value = '';
        this.selected = { option: undefined, isSelected: false };
        this._panelOpen = false;
        this.identifier = `select-${MasSelect.selectCount++}`;
        this.disabled = false;
        this.invalid = false;
        this.size = 'S';
        this.filled = false;
        this.change = new EventEmitter();
        this._destroy = new Subject();
        this.optionChanges = defer(() => {
            const options = this.options;
            if (options) {
                return options.changes.pipe(startWith(options), switchMap(() => merge(...options.map((option) => option.onChange))));
            }
            return this._ngZone.onStable.pipe(take(1), switchMap(() => this.optionChanges));
        });
        this.onChangeHandler = (_) => { };
        this.onTouchedHandler = () => { };
    }
    clickOut(event) {
        if (!this.eRef.nativeElement.contains(event.target)) {
            this.close();
        }
    }
    _resetOptions() {
        const changedOrDestroyed = merge(this.options.changes, this._destroy);
        this.optionChanges.pipe(takeUntil(changedOrDestroyed)).subscribe((event) => {
            this._onSelect(event.option);
            if (this._panelOpen) {
                this.close();
                this.focusOut();
            }
        });
    }
    _onSelect(option) {
        if (option.value == null) {
            option.deselect();
        }
        else if (!option.selected) {
            this.options.map((option) => {
                if (option.selected)
                    option.deselect();
            });
            option.select();
            this.selected.isSelected = true;
            this.selected.option = this.options.filter((option) => option.selected)[0];
        }
    }
    toggle() {
        this.panelOpen ? this.close() : this.open();
    }
    open() {
        this._panelOpen = true;
    }
    close() {
        this._panelOpen = false;
    }
    focus() { }
    onChange(event) {
        this.onChangeHandler(event.target.value);
        this.change.emit(event.target.value);
    }
    focusOut() {
        this.onTouchedHandler();
    }
    isTemplate(value) {
        return value instanceof TemplateRef;
    }
    writeValue(obj) {
        this.value = obj;
    }
    registerOnChange(fn) {
        this.onChangeHandler = fn;
    }
    registerOnTouched(fn) {
        this.onTouchedHandler = fn;
    }
    setDisabledState(isDisabled) {
        this.disabled = isDisabled;
    }
}
MasSelect.selectCount = 0;
MasSelect.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: MasSelect, deps: [{ token: i0.NgZone }, { token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component });
MasSelect.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.2.9", type: MasSelect, selector: "mas-select", inputs: { label: "label", identifier: "identifier", disabled: "disabled", invalid: "invalid", size: "size", filled: "filled", value: "value", placeholder: "placeholder" }, outputs: { change: "change" }, host: { listeners: { "document:click": "clickOut($event)" }, properties: { "class.mas-select--invalid": "invalid", "class.mas-select--disabled": "disabled", "class.mas-select--medium": "size === \"M\"", "class.mas-select--small": "size === \"S\"" }, classAttribute: "mas-select" }, providers: [
        {
            provide: NG_VALUE_ACCESSOR,
            useExisting: MasSelect,
            multi: true,
        },
    ], queries: [{ propertyName: "options", predicate: MasOption }], ngImport: i0, template: "<ng-content select=\"mas-label\"></ng-content>\r\n<div\r\n  class=\"mas-select_value-wrapper\"\r\n  [ngSwitch]=\"empty\"\r\n  (click)=\"toggle()\"\r\n  (change)=\"onChange($event)\"\r\n  [ngClass]=\"{ 'mas-select_listbox--opened': panelOpen }\"\r\n>\r\n  <i class=\"mas-warningcircle-outlined mas-icon\" *ngSwitchCase=\"true\"></i>\r\n  <span\r\n    class=\"mas-select-placeholder mas-select-min-line\"\r\n    *ngSwitchCase=\"true\"\r\n    [ngClass]=\"{ 'mas-select--filled': filled }\"\r\n    >{{ placeholder }}</span\r\n  >\r\n  <span class=\"mas-select_text mas-select-min-line\" *ngSwitchCase=\"false\">\r\n    {{ triggerValue }}\r\n  </span>\r\n  <i class=\"mas-caretdown-outlined mas-icon\"></i>\r\n</div>\r\n<!-- ! listbox -->\r\n<ng-template [ngIf]=\"panelOpen && !disabled\">\r\n    <div #select role=\"listbox\" [ngClass]=\"{ 'mas-select--disabled': disabled }\" class=\"mas-select_listbox-wrapper\">\r\n        <ng-content select=\"mas-option\"></ng-content>\r\n    </div>\r\n</ng-template>\r\n<ng-content *ngIf=\"invalid\" select=\"mas-error\"></ng-content>\r\n<ng-content *ngIf=\"!invalid\" select=\"mas-helper\"></ng-content>\r\n", dependencies: [{ kind: "directive", type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "directive", type: i1.NgSwitch, selector: "[ngSwitch]", inputs: ["ngSwitch"] }, { kind: "directive", type: i1.NgSwitchCase, selector: "[ngSwitchCase]", inputs: ["ngSwitchCase"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: MasSelect, decorators: [{
            type: Component,
            args: [{ selector: 'mas-select', providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: MasSelect,
                            multi: true,
                        },
                    ], host: {
                        class: 'mas-select',
                        '[class.mas-select--invalid]': 'invalid',
                        '[class.mas-select--disabled]': 'disabled',
                        '[class.mas-select--medium]': 'size === "M"',
                        '[class.mas-select--small]': 'size === "S"'
                    }, template: "<ng-content select=\"mas-label\"></ng-content>\r\n<div\r\n  class=\"mas-select_value-wrapper\"\r\n  [ngSwitch]=\"empty\"\r\n  (click)=\"toggle()\"\r\n  (change)=\"onChange($event)\"\r\n  [ngClass]=\"{ 'mas-select_listbox--opened': panelOpen }\"\r\n>\r\n  <i class=\"mas-warningcircle-outlined mas-icon\" *ngSwitchCase=\"true\"></i>\r\n  <span\r\n    class=\"mas-select-placeholder mas-select-min-line\"\r\n    *ngSwitchCase=\"true\"\r\n    [ngClass]=\"{ 'mas-select--filled': filled }\"\r\n    >{{ placeholder }}</span\r\n  >\r\n  <span class=\"mas-select_text mas-select-min-line\" *ngSwitchCase=\"false\">\r\n    {{ triggerValue }}\r\n  </span>\r\n  <i class=\"mas-caretdown-outlined mas-icon\"></i>\r\n</div>\r\n<!-- ! listbox -->\r\n<ng-template [ngIf]=\"panelOpen && !disabled\">\r\n    <div #select role=\"listbox\" [ngClass]=\"{ 'mas-select--disabled': disabled }\" class=\"mas-select_listbox-wrapper\">\r\n        <ng-content select=\"mas-option\"></ng-content>\r\n    </div>\r\n</ng-template>\r\n<ng-content *ngIf=\"invalid\" select=\"mas-error\"></ng-content>\r\n<ng-content *ngIf=\"!invalid\" select=\"mas-helper\"></ng-content>\r\n" }]
        }], ctorParameters: function () { return [{ type: i0.NgZone }, { type: i0.ElementRef }]; }, propDecorators: { label: [{
                type: Input
            }], identifier: [{
                type: Input
            }], disabled: [{
                type: Input
            }], invalid: [{
                type: Input
            }], size: [{
                type: Input
            }], filled: [{
                type: Input
            }], value: [{
                type: Input
            }], placeholder: [{
                type: Input
            }], change: [{
                type: Output
            }], options: [{
                type: ContentChildren,
                args: [MasOption]
            }], clickOut: [{
                type: HostListener,
                args: ['document:click', ['$event']]
            }] } });
