// @masoni/components/dist/esm2020/lib/select/select.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { MasErrorModule, MasHelperModule, MasLabelModule } from '../sub-elements';
import { MasSelect } from './select.component';
import { MasOption } from './option.component';
import { MasOptionGroup } from './option-group.directive';
import * as i0 from "@angular/core";
export class MasSelectModule {
}
MasSelectModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: MasSelectModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MasSelectModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.9", ngImport: i0, type: MasSelectModule, declarations: [MasSelect, MasOption, MasOptionGroup], imports: [CommonModule, FormsModule, MasErrorModule, MasHelperModule, MasLabelModule], exports: [MasSelect, MasOption, MasOptionGroup] });
MasSelectModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: MasSelectModule, imports: [CommonModule, FormsModule, MasErrorModule, MasHelperModule, MasLabelModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: MasSelectModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MasSelect, MasOption, MasOptionGroup],
                    imports: [CommonModule, FormsModule, MasErrorModule, MasHelperModule, MasLabelModule],
                    exports: [MasSelect, MasOption, MasOptionGroup],
                }]
        }] });
