// @masoni/components/dist/esm2020/lib/text-input/text-input.module.mjs
import { NgModule } from '@angular/core';
import { MasTextInput } from './text-input.component';
import { CommonModule } from '@angular/common';
import { MasErrorModule, MasLabelModule, MasHelperModule } from '../sub-elements';
import { ReactiveFormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
export class MasTextInputModule {
}
MasTextInputModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: MasTextInputModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MasTextInputModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.9", ngImport: i0, type: MasTextInputModule, declarations: [MasTextInput], imports: [CommonModule, MasErrorModule, MasLabelModule, MasHelperModule, ReactiveFormsModule], exports: [MasTextInput] });
MasTextInputModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: MasTextInputModule, imports: [CommonModule, MasErrorModule, MasLabelModule, MasHelperModule, ReactiveFormsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: MasTextInputModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, MasErrorModule, MasLabelModule, MasHelperModule, ReactiveFormsModule],
                    declarations: [MasTextInput],
                    exports: [MasTextInput],
                    providers: [],
                }]
        }] });
